src/coreclr/System.Private.CoreLib/src/System/Reflection/LoaderAllocator.cs

```diff
@@ -34,6 +34,8 @@ internal sealed partial class LoaderAllocatorScout
             if (m_nativeLoaderAllocator == IntPtr.Zero)
                 return;
 
+            VirtualDispatchHelpers.ClearCache();
```

Comment:
There may be a subtle race condition:
1. the loader allocator is still in use when `VirtualDispatchHelpers.ClearCache` run
2. the thread is rescheduled right `VirtualDispatchHelpers.ClearCache`
3. the loader allocator becomes unreferenced
4. Destroy succeeds

I think we will end up with stale entries in the hashtable.

I am wondering whether this is the reason for the casting cache flush to be done while the runtime is suspended: https://github.com/dotnet/runtime/pull/106843/files#diff-060f68aac95f40dd380ee5a82dd047222232b25646f908c1ea394438e18a8709R604